src/Storage/EntityManager.php

```diff
@@ -262,42 +268,54 @@ public function getRepository($className)
         if (array_key_exists($classMetadata->getName(), $this->repositories)) {
             $repoClass = $this->repositories[$classMetadata->getName()];
             if (is_callable($repoClass)) {
-                return call_user_func_array($repoClass, [$this, $classMetadata]);
+                $repo = call_user_func_array($repoClass, [$this, $classMetadata]);
             }
 
-            return new $repoClass($this, $classMetadata);
+            $repo = $repoClass($this, $classMetadata);
         }
 
-        foreach ($this->aliases as $alias => $namespace) {
-            $full = str_replace($alias, $namespace, $className);
+        if (is_null($repo)) {
+            foreach ($this->aliases as $alias => $namespace) {
+                $full = str_replace($alias, $namespace, $className);
 
-            if (array_key_exists($full, $this->repositories)) {
-                $classMetadata = $this->getMapper()->loadMetadataForClass($full);
-                $repoClass = $this->repositories[$full];
+                if (array_key_exists($full, $this->repositories)) {
+                    $classMetadata = $this->getMapper()->loadMetadataForClass($full);
+                    $repoClass = $this->repositories[$full];
 
-                return new $repoClass($this, $classMetadata);
+                    $repo = new $repoClass($this, $classMetadata);
+                }
             }
         }
 
+
         /*
          * The metadata driver can also attempt to resolve an alias for us.
          * For now we are hardcoding the link between a content entity and
          * the content repository, but in time this should be a metadata level
          * configuration.
          */
-        if ($this->getMapper()->resolveClassName($className) === 'Bolt\Storage\Entity\Content') {
-            return $this->getDefaultRepositoryFactory($classMetadata);
+        if (is_null($repo) && $this->getMapper()->resolveClassName($className) === 'Bolt\Storage\Entity\Content') {
```

Comment:
Small style consistency request… can you use comparison rather than functions:
```
if ($repo === null && $this->getMapper()->resolveClassName($className) === 'Bolt\Storage\Entity\Content') {
```

We're pretty much locked in on PSR-2 + [Symfony Coding Standards](http://symfony.com/doc/current/contributing/code/standards.html) … the main difference is that we still use spaces around concatenation operators.